functions/consumer/s3-consumer/src/main/java/org/springframework/cloud/fn/consumer/s3/AwsS3ConsumerEndpointConfiguration.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2020-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.fn.consumer.s3;
+
+import com.amazonaws.client.builder.AwsClientBuilder.EndpointConfiguration;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.aws.core.region.RegionProvider;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * @author Timo Salm
+ */
+@Configuration
+public class AwsS3ConsumerEndpointConfiguration {
+
+	@Bean
+	@ConditionalOnProperty("s3.consumer.endpoint-url")
```

Comment:
Huh? Why this cannot be as a conditional bean in the common `AmazonS3Configuration`?
Why do we need to duplicate it? 
Right we probably need to introduce an `S3ClientProperties`, but then we just don't have any changes in the `supplier`, nor `consumer`.